src/components/button-link/index.tsx

```diff
@@ -57,9 +57,11 @@ const ButtonLink = ({
 			opensInNewTab={opensInNewTab}
 			rel={opensInNewTab ? 'noreferrer noopener' : undefined}
 		>
-			{hasLeadingIcon && icon}
-			{hasText ? text : null}
-			{hasTrailingIcon && icon}
+			<>
+				{hasLeadingIcon && icon}
+				{hasText ? text : null}
+				{hasTrailingIcon && icon}
+			</>
```

Comment:
The types were reporting that `Link` only accepts a single child, so I wrapped in a fragment here.